### PR TITLE
Use k8s svc instead of pod IP in catalogsource.bash

### PR DESCRIPTION
**Pull request by maschmid**
The `wait_until_pods_running` and `indexip` combo doesn't always work, as sometimes the pod isn't even created yet, so `wait_until_pods_running` doesn't wait for anything and `indexip` command fails on no item in `items[0]` (as no such pod exists yet)

Let's change it to just use a k8s svc instead...

### CODE REVIEW

File: hack/lib/catalogsource.bash [hunk 1]
```diff
@@ -96,7 +107,7 @@ kind: CatalogSource
 metadata:
   name: serverless-operator
 spec:
-  address: $indexip:50051
+  address: serverless-index.openshift-marketplace.svc:50051
```
Comment (maschmid) [inline hack/lib/catalogsource.bash:110]: oops, should not hardcode OLM_NAMESPACE...

**Review by cardil — APPROVED** [review]
Much more sane approach

/lgtm

**Comment by markusthoemmes** [comment]
/hold

There was a reason I did this. IIRC, the source had trouble coming up if I didn't wait for the pod to be up first and that never reconciled (or at least not in a usual amount of time). Can we please doublecheck that?

File: hack/lib/catalogsource.bash [hunk 2]
```diff
@@ -85,9 +85,20 @@ spec:
           /bin/opm registry serve -d index.db -p 50051
 EOF
 
-  # Wait for the index pod to be up to avoid inconsistencies with the catalog source.
-  wait_until_pods_running "$OLM_NAMESPACE"
```
Comment (markusthoemmes) [inline hack/lib/catalogsource.bash:89]: Do we still need to wait somewhere for the CatalogSource itself to become ready? The pod takes quite a while to come up. Maybe it'd be a compromise to wait for the Available condition on the deployment.

**Comment by maschmid** [comment]
obsolete by https://github.com/openshift-knative/serverless-operator/pull/622

**Pull request closed without merging**